dbt-tests-adapter/dbt/tests/adapter/basic/test_get_catalog_for_single_relation.py

```diff
@@ -0,0 +1,123 @@
+import pytest
+
+from dbt.adapters.capability import Capability
+from dbt_common.contracts.metadata import (
+    TableMetadata,
+    StatsItem,
+    CatalogTable,
+    ColumnMetadata,
+)
+from dbt.tests.util import run_dbt, get_connection
+
+models__my_model_sql = """
+{{
+    config(
+        materialized='view',
+    )
+}}
+
+select * from {{ ref('my_seed') }}
+"""
+
+seed__my_seed_csv = """id,first_name,email,ip_address,updated_at
+1,Larry,lking0@miitbeian.gov.cn,69.135.206.194,2008-09-12 19:08:31
+"""
+
+
+class BaseGetCatalogForSingleRelation:
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {"name": "get_catalog_for_single_relation"}
+
+    @pytest.fixture(scope="class")
+    def seeds(self):
+        return {
+            "my_seed.csv": seed__my_seed_csv,
+        }
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "my_model.sql": models__my_model_sql,
+        }
+
+    @pytest.fixture(autouse=True)
+    def clean_up(self, project):
```

Comment:
I thought we only needed this when we create a schema other than the default test schema on the project fixture. I don't think it hurts anything to keep it here, but I also don't think we need it. If you do keep it, I would add `scope="class"` to the fixture. It doesn't matter at the moment since there is only one test method in this test class. But if we add a second, we can save ourselves some time by accounting for it now.